include/irr/asset/IAsset.h

```diff
@@ -173,14 +173,24 @@ class IAsset : virtual public core::IReferenceCounted
 
         virtual core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const = 0;
 
-		//! restore() is not recursive!
-		virtual bool restore(IAsset* _other)
+		bool restoreFromDummy(IAsset* _other, uint32_t _levelsBelow)
 		{
-			if (!this->isADummyObjectForCache())
+			if (!canBeRestoredFrom(_other))
+				return false;
+
+			restoreFromDummy_impl(_other, _levelsBelow);
+			return true;
+		}
+
+		virtual bool canBeRestoredFrom(const IAsset* _other) const
+		{
+			if (getMutability() != EM_MUTABLE)
+				return false;
+			if (!isADummyObjectForCache())
+				return false;
```

Comment:
this should not be here, just because asset is not mutable or is not a dummy object (cannot be restored, doesn't matter what `_other` is) does not meen the dependencies couldn't be restored.

Same if `_other` is a dummy object, the dependants of `this` which are mutable and dummy could be restored from dependants of `_other` which are mutable and non-dummy.

**However if `_other` does not "match"/"is not compatible" with `this` (i.e. different layout contents) then obviously there's no point restoring the dependants.** 

P.S. when restoring you should check that `_other` is mutable.